src/NodaTime/Calendars/WondrousYearMonthDayCalculator.cs

```diff
@@ -0,0 +1,385 @@
+﻿// Copyright 2014 The Noda Time Authors. All rights reserved.
+// Use of this source code is governed by the Apache License 2.0,
+// as found in the LICENSE.txt file.
+
+using NodaTime.Calendars.Wondrous;
+using NodaTime.Utility;
+using System;
+
+namespace NodaTime.Calendars
+{
+    /// <summary>
+    ///     See <see cref="CalendarSystem.GetWondrousCalendar" /> for details.
+    /// </summary>
+    internal sealed class WondrousYearMonthDayCalculator : YearMonthDayCalculator
+    {
+        private const int WondrousMinYear = 1;
+        private const int WondrousMaxYear = 1000;
```

Comment:
Is this arbitrary, or inherently part of the calendar?

Reply:
Arbitrary. I'm not aware of precise guidelines for before year 1.  The embedded knowledge (below) is only defined to the year 1000, so I don't want to go beyond it for now. That is still 800 years in our future, so there should be time for an update before it becomes a serious concern!